integrations/optimizely/lib/index.js

```diff
@@ -213,6 +214,22 @@ Optimizely.prototype.sendClassicDataToSegment = function(experimentState) {
       variationName: variationNames.join(', ') // eg. 'Variation X' or 'Variation 1, Variation 2'
     };
 
+    // If customProps is provided overide the props with it.
+    // If valid customProps present it will override existing props.
+    var customProps = this.options.customProps;
+    var customPropsKeys = Object.keys(customProps);
+    var data = window.optimizely && window.optimizely.data;
+
+    if (data && customPropsKeys.length) {
+      for (var index = 0; index < customPropsKeys.length; index++) {
+        var segmentProp = customPropsKeys[index];
+        var optimizelyProp = customProps[segmentProp];
+        if (data[optimizelyProp] !== 'undefined') {
```

Comment:
Would this be instead `data[optimizelyProp] !== undefined`? (without the quotes).

Reply:
Missed `typeof` here, good catch @CarlosMecha